auth.js

```diff
@@ -0,0 +1,154 @@
+// Auth 활성화
+const form = document.querySelector("form");
+const authBt = document.querySelector(".auth-bt");
+const loginBt = document.querySelector(".auth-bt.login");
+const joinBt = document.querySelector(".auth-bt.join");
+const emailInput = document.querySelector(".input-contain .email");
+const passwordInput = document.querySelector(".input-contain .password");
+const passwordConfirmInput = document.querySelector(
+  ".input-contain .password-confirm"
+);
+
+form.addEventListener("input", () => {
+  authBt.disabled = !form.checkValidity();
+});
+
+const handleLoginClick = (e) => {
+  e.preventDefault();
+  window.location.href = "/items.html";
+};
+
+if (loginBt) {
+  loginBt.addEventListener("click", handleLoginClick);
+}
+
+const handleJoinClick = (e) => {
+  e.preventDefault();
+  window.location.href = "/login.html";
+};
+
+if (joinBt) {
+  joinBt.addEventListener("click", handleJoinClick);
+}
+
+// input 이메일 error 처리
+const emailNoneError = document.querySelector(".error-message.email-none");
+const emailWrongError = document.querySelector(".error-message.email-wrong");
+
+const handleEmailFocusout = (e) => {
+  const value = e.target.value.trim();
+  if (value.length === 0) {
+    emailNoneError.classList.add("show");
+    emailInput.classList.add("error-box");
+    emailWrongError.classList.remove("show");
+  } else if (!emailInput.checkValidity()) {
+    emailNoneError.classList.remove("show");
+    emailInput.classList.add("error-box");
+    emailWrongError.classList.add("show");
+  } else {
+    emailNoneError.classList.remove("show");
+    emailWrongError.classList.remove("show");
+    emailInput.classList.remove("error-box");
+  }
+};
+emailInput.addEventListener("focusout", handleEmailFocusout);
+
+// input 비밀번호 error 처리
+const passwordNoneError = document.querySelector(
+  ".error-message.password-none"
+);
+const passwordWrongError = document.querySelector(
+  ".error-message.password-wrong"
+);
+
+let passwordValue = "";
+
+const handlePasswordFocusout = (e) => {
+  passwordValue = e.target.value.trim();
+  if (passwordValue.length === 0) {
+    passwordNoneError.classList.add("show");
+    passwordInput.classList.add("error-box");
+    passwordWrongError.classList.remove("show");
+  } else if (!passwordInput.checkValidity()) {
+    passwordNoneError.classList.remove("show");
+    passwordInput.classList.add("error-box");
+    passwordWrongError.classList.add("show");
+  } else {
+    passwordNoneError.classList.remove("show");
+    passwordWrongError.classList.remove("show");
+    passwordInput.classList.remove("error-box");
+  }
+};
```

Comment:
💊 제안
passwordValue 변수를 함수 바깥에 두기보다는, 함수 내부에서 선언하는 것이 더 명확할 것 같습니다.
하단의 handlePasswordConfirm 함수에서 재사용하기 위해 이렇게 작성하신 것 같지만,
필요한 시점에 값을 직접 읽어오는 방식이 더 직관적이고 유지보수에도 유리합니다~

```suggestion
const handlePasswordFocusout = (e) => {
  const passwordValue = e.target.value.trim();
  
  if (passwordValue.length === 0) {
    passwordNoneError.classList.add("show");
    passwordInput.classList.add("error-box");
    passwordWrongError.classList.remove("show");
  } else if (!passwordInput.checkValidity()) {
    passwordNoneError.classList.remove("show");
    passwordInput.classList.add("error-box");
    passwordWrongError.classList.add("show");
  } else {
    passwordNoneError.classList.remove("show");
    passwordWrongError.classList.remove("show");
    passwordInput.classList.remove("error-box");
  }
};
```